report_thesis/src/sections/background/ensemble_learning_models/xgboost.tex

```diff
@@ -1 +1,27 @@
-\subsubsection{XGBoost}
\ No newline at end of file
+\subsubsection{XGBoost}
+We now describe XGBoost based on \citet{ChenGuestrin2016}.
+
+\gls{xgboost} is an advanced implementation of gradient boosting designed for speed and performance. 
+It builds upon traditional boosting techniques by introducing several algorithmic enhancements to improve execution speed and model performance.
+
+The core mechanism of \gls{xgboost} can be summarized in the following way:
+\begin{enumerate}
+\item The algorithms first step involves sorting the feature values and proposing candidate split points based on the distribution of these values.
+\item It then maps these features into buckets based on the candidate splitting points.
+\item For each candidate split, the algorithm calculates a split score using a loss function that measures the difference between predicted and actual values.
+\item The split that minimizes the loss function is chosen, and the data is partitioned based on this optimal split point. 
+	\begin{itemize}
+		\item Parallelism is leveraged here, where the computations for different features and split points can be distributed across multiple processors. 
+		This is facilitated by storing each feature in a compressed, sorted column format, allowing efficient linear scans and parallel processing.
+	\end{itemize}
+\item This process is recursively applied to each partition, building a tree structure by continually splitting the data to minimize the loss function at each node.
+\end{enumerate}
```

Comment:
I think this should be prose. It's written in prose (e.g. "the algorithm's first step"), but it's listed in a numerical list.

Generally, I have tried to avoid writing lists in the background section, instead opting to write it in prose and splitting steps with new lines.